weblate/static/editor/full.js

```diff
@@ -36,6 +36,32 @@
       submitForm({ target: this.$translationArea });
     });
 
+    /* Delete machinery results */
+    this.$editor.on("click", ".js-delete-machinery", (e) => {
+      var $el = $(e.target);
+
+      /* Delete Url dialog */
+      var $deleteEntriesDialog = null;
+      this.$editor.on("show.bs.modal", "#delete-url-modal", (e) => {
+        $deleteEntriesDialog = $(e.currentTarget);
+        var text = $el.parent().parent().data("raw").text;
+        var modalBody = this.machinery.renderDeleteUrls(text);
+        $deleteEntriesDialog.find(".modal-body").append(modalBody);
+      });
+
+      this.$editor.on("submit", ".delete-url-form", (e) => {
+        var $form = $(e.currentTarget);
+        var $deleteEntries = $form.find("input.form-check-input:checked");
+        $deleteEntriesDialog.modal("hide");
+
+        Object.entries($deleteEntries).forEach(([_, entry]) => {
+          if (typeof entry.id !== "undefined") {
+            this.removeTranslationEntry(entry.id);
+          }
+        });
+      });
```

Comment:
You need to supress further event processing here, otherwise the form is submitted and the AJAX request will get terminated:

```suggestion
      return false;
      });
```